s3fs/core.py

```diff
@@ -919,6 +919,13 @@ async def _makedirs(self, path, exist_ok=False):
     makedirs = sync_wrapper(_makedirs)
 
     async def _rmdir(self, path):
+        bucket, key, _ = self.split_path(path)
+        if key:
+            if await self._exists(path):
+                # User may have meant rm(path, recursive=True)
+                raise FileExistsError
```

Comment:
By doing this, we are explicitly not dealing with directory placeholder files. Perhaps that should be stated in the docstring.